### PR TITLE
PR : Release SPAShip

### PR DESCRIPTION
**Pull request by soumyadip007**

### CODE REVIEW

**Comment by arkaprovob** [comment]
I will take a look and merge.

**Comment by arkaprovob** [comment]
@soumyadip007 setup instruction is missing, Please modify the readme file remove existing content and just add basic instructions to set up and run this app

**Comment by soumyadip007** [comment]
@arkaprovob , I've modified the README.md. We need the Router and Manager UI using that.

**Review by arkaprovob — CHANGES_REQUESTED** [review]
Also mention 

- the usage of the router and common package
- please add more details on setup CLI and spaship-manager separately .
-  there are readme files inside individual packages as well. Would you please check them and make changes accordingly?

**Review by arkaprovob — CHANGES_REQUESTED** [review]
Please check my comments and resolve them.

File: packages/manager-ui/src/components/property/NewRepositoryConfigs.tsx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+  const [token, setToken] = useState("");
+  const [filePathRequest, setFilePathRequest] = useState<any[]>([]);
+
+  const [treePath, setTreePath] = useState<any[]>([]);
+  const [filePath, setFilePath] = useState<any[]>([]);
+  const [event, setEvent] = useState<any[]>([]);
+
+  useEffect(() => {
+    if (repositoryLink.length > 23) {
+      axios.get(`https://api.github.com/repos/soumyadipXD/spaship-spas/branches`)
```
[the remark this answers is not in the document]
Comment (soumyadip007) [reply packages/manager-ui/src/components/property/NewRepositoryConfigs.tsx:26]: NewRepositoryConfig was an redundant file, removed from the project.

File: packages/manager/controllers/websites/gitOperations/gitOperations.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                        "Created new-branch on HEAD");
+                });
+        });
+
+    console.log(`Created Branch : ${localBranch}`);
+    await delay(100);
+}
+
+function createSignature() {
+    return Git.Signature.now("Soumyadip Chowdhury",
```
[the remark this answers is not in the document]
Comment (soumyadip007) [reply packages/manager/controllers/websites/gitOperations/gitOperations.js:195]: I've added spaship-deployment as the signature

File: packages/manager/controllers/websites/gitOperations/gitOperations.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        fs.writeFileSync(`${pathFile}${spa.spaName}/.spaship`, JSON.stringify(spaShipFile, null, "\t"));
+        console.log(`./spaship added at ${pathFile}${spa.spaName}`)
+    }
+}
+
+async function gitOperationsCommit(repository, signature, resolvePathCreateBranch, localBranch) {
+    let referr;
+    let index;
+    let oid;
+    let remoteGitURL = "ssh://git@xxxxxx/home/git/git_test_repo.git";
```
[the remark this answers is not in the document]
Comment (soumyadip007) [reply packages/manager/controllers/websites/gitOperations/gitOperations.js:102]: Unused variables, removed.

File: packages/manager/controllers/websites/gitOperations/gitOperations.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+        .then((repo) => {
+            return repo.getHeadCommit()
+                .then((targetCommit) => {
+                    return repo.createBranch(remoteBranch, targetCommit, false);
+                })
+                .then((reference) => {
+                    return repo.checkoutBranch(reference, {});
+                })
+                .then(() => {
+                    return repo.getReferenceCommit('refs/remotes/origin/' + remoteBranch);
```
[the remark this answers is not in the document]
Comment (soumyadip007) [reply packages/manager/controllers/websites/gitOperations/gitOperations.js:50]: It'll work for commits. Sure

File: packages/manager/controllers/websites/gitOperations/gitOperations.js [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+    await cloneGitRepository(req.body?.repositoryConfigs[0].repositoryLink, pathClone);
+    await checkoutRemoteBranch(req.body?.repositoryConfigs[0].branch, resolvePathCreateBranch);
+    await gitCreateBranch(resolvePathCreateBranch, localBranch);
+    repository = await gitCheckout(repository, resolvePathCreateBranch, localBranch);
+    await createSPAShipTemplateRequest(req, pathFile);
+    await gitOperationsCommit(repository, signature, resolvePathCreateBranch, localBranch);
+    await zipFiles(directoryName);
+    const websiteResponse = await saveWebsite(req, res);
+    res.send({ actionStatus: "Git Actions Performed Successfully", path: path.resolve(__dirname, `./../../../root/${directoryName}.zip`), websiteResponse : websiteResponse });
```
[the remark this answers is not in the document]
Comment (soumyadip007) [reply packages/manager/controllers/websites/gitOperations/gitOperations.js:30]: const basePath = config.get("directoryBasePath");
const resolvePathCreateBranch = `../../../${basePath}/${directoryName}/.git`;

Reading from ENV.

File: packages/manager/controllers/websites/gitOperations/gitOperations.js [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+async function zipFiles(directoryName) {
+    //await delay(100);
+    await zip(path.resolve(__dirname, `./../../../root/${directoryName}`), path.resolve(__dirname, `./../../../root/${directoryName}.zip`));
```
[the remark this answers is not in the document]
Comment (arkaprovob) [reply packages/manager/controllers/websites/gitOperations/gitOperations.js:37]: @soumyadip007 save it in a named directory example /apps/zip and add an env support for the location

**Comment by arkaprovob** [comment]
Everything looks good, in next to get rid of all dotted directory navigation.

**Pull request merged**